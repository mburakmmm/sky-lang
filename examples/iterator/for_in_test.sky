# For...in iterator protocol test

function main
  print("=== List Iteration ===")
  let numbers = [10, 20, 30]
  for num in numbers
    print(num)
  end
  
  print("=== Dict Iteration ===")
  let person = {"name": "Bob", "age": "25"}
  for key in person
    print(key)
  end
  
  print("=== String Iteration ===")
  let text = "SKY"
  for char in text
    print(char)
  end
  
  print("=== Done ===")
end

