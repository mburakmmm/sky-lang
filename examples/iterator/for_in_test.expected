=== List Iteration ===
10
20
30
=== Dict Iteration ===
age
name
=== String Iteration ===
S
K
Y
=== Done ===

